### DISCUSSION TITLE
Template tags

### DISCUSSION THREAD

**Discussion by richcorbs** [pinecone-router/router#6] [Q&A] [answered]
Is having some of the configuration in template tags significant?

Likewise, could all of the configuration accomplished by HTML attributes be accomplished in javascript?

**Comment by rehhouari** [comment]
Which configuration exactly? The only configs that are in template tags are: `x-route` `x-handler` and/or `x-view`, which are specific to the *route itself*.
Other configuration like are set on the *component element* (one with `x-data`) which include `x-hash` `x-base` and those are specific to *the entire router*.

There are some configuration you can set with javascript indeed and they're mentioned in the [README].(https://github.com/rehhouari/alpine-router#settings)

If you're asking for the reason why the route config (x-route, xhandler)  are on the template tags and not done using javascript like:
`routes({'/home', hanlder})` Is because there are already javascript routers that does exactly that like [page.js](https://github.com/visionmedia/page.js).

This is a router specifically for those using Alpine.js and like its syntax, which don't involve javascript and make use of template tags and attributes :)

And last, no, not all configs accomplished by HTML attributes can be configured by Javascript, _yet_.
Reason is some configs must be set *before router is loaded*. Meaning you need to configure the router with javascript *before* importing Alpine Router.

I'll consider allowing that, the syntax would be like:
```javascript
window.AlpineRouterConfig  = {interceptLinks: false, ....}
```

**Reply by rehhouari** [reply]
oh and If you have any suggestions on how it should be done feel free to tell me ^^

**Comment by rehhouari** [comment, marked as answer]
Added [Javascript settings!](https://github.com/pinecone-router/router#settings)
Removed some HTML attribute settings with these; not the ones in templates tho.
